### PR TITLE
Cleanup docstrings for `gammapy.analysis`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue https://github.com/gammapy/gammapy/issues/4811, specifically for `gammapy.analysis`

### CODE REVIEW

**Review by cgalelli — APPROVED** [review]
Thanks @Astro-Kirsty! Seems fine to me!

File: gammapy/analysis/core.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class Analysis:
     Parameters
     ----------
     config : dict or `AnalysisConfig`
```
Comment (registerrier) [inline gammapy/analysis/core.py:46]:
```suggestion
    config : dict or `~gammapy.analysis.AnalysisConfig`
```

File: gammapy/analysis/core.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def models(self, models):
 
     @property
     def config(self):
-        """Analysis configuration (`AnalysisConfig`)"""
+        """Analysis configuration as an `AnalysisConfig` object."""
```
Comment (registerrier) [inline gammapy/analysis/core.py:78]:
```suggestion
        """Analysis configuration as an `~gammapy.analysis.AnalysisConfig` object."""
```

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty ! Looks good.

**Pull request merged**